### ISSUE TITLE
Webtrees shows JSON text instead of rendered HTML after creating a new shared place

### ISSUE THREAD

**Issue by Jefferson49** [vesta-webtrees-2-custom-modules/vesta_shared_places#159]
Observed with webtrees 2.1.16 and vesta 2.1.16.1.5

Sometimes, I get the following response in the browser, after creating a new shared place. The output is shown as text instead of as rendered HTML:

{"value":"@P10552@","text":"Syracuse, Onondaga County, New York, USA","title":"Syracuse, Onondaga County, New York, USA","html":"\n<div class=\"modal-header\">\n    <h3 class=\"modal-title\" id=\"wt-ajax-modal-title\">\n        Der gemeinsame Ort <span dir=\"auto\">Syracuse<\/span> wurde erstellt.    <\/h3>\n    <button type=\"button\" class=\"btn-close\" data-bs-dismiss=\"modal\" aria-label=\"Schließen\">\n    <\/button>\n<\/div>\n\n<div class=\"modal-body\">\n    <a href=\"http:\/\/nas-synology220\/webtrees\/index.php?route=%2Fwebtrees%2Ftree%2Fhemprich%2FsharedPlace%2FP10552%2FSyracuse\">\n        <span dir=\"auto\">Syracuse<\/span>  (Hinweis: Es wurde außerdem ein gemeinsamer Ort auf einer höheren Ebene angelegt)    <\/a>\n<\/div>\n\n\n<div class=\"modal-footer\">\n    <button type=\"button\" class=\"btn btn-primary\" data-bs-dismiss=\"modal\">\n        <span class=\"wt-icon-cancel\"><i class=\"fa-solid fa-times fa-fw\" aria-hidden=\"true\"><\/i><\/span>        Schließen    <\/button>\n<\/div>\n"}

Related URL:
../webtrees/index.php?route=%2Fwebtrees%2Ftree%2Fhemprich%2Fcreate-location

I do not know how to reproduce this behavior. It only occurs occationately in rare cases.

**Comment by Jefferson49** [comment]
Maybe this behavior is related or simular to this topic:
[SyntaxError: Unexpected token '<', " on add birth location](https://www.webtrees.net/index.php/en/forum/help-for-release-2-1-x/38029-syntaxerror-unexpected-token-on-add-birth-location)

In my webtrees installation, I have already included the related bugfix from Greg:
[Bugfix commit d5c0e14](https://github.com/fisharebest/webtrees/commit/d5c0e143ab185dce53ea5501376ff09775c6a4a6)

In the past, I also observed the "SyntaxError: Unexpected token '<'" error like described in the forum post. After using Greg's bugfix, I have not observed this error again. However, the JSON text behavior occured also with the bugfix installed.

**Comment by ric2016** [comment]
The webtrees bugfix is likely irrelevant here, Vesta doesn't use that action.

The json output is shown in the modal dialog?
From where do you create the shared place when this error occurs?
(Note that apparently the confirmation dialog is no longer shown always, only when creating from the shared places list. I don't think this was done intentionally. I'll probably have to fix that first)

**Comment by Jefferson49** [comment]
> The json output is shown in the modal dialog?

I think it was shown directly in the browser window and not in the modal. However, I am not 100% sure.

> From where do you create the shared place when this error occurs?

When editing a birth and creating a new shared place for the birth place.

![Syracuse](https://github.com/vesta-webtrees-2-custom-modules/vesta_shared_places/assets/81484983/d8a30e8f-5758-4907-9947-487839c5487b)
![Syracuse2](https://github.com/vesta-webtrees-2-custom-modules/vesta_shared_places/assets/81484983/0bc4283e-9ae2-4901-9c16-3fec73b560fc)

**Comment by ric2016** [comment]
> I think it was shown directly in the browser window and not in the modal. However, I am not 100% sure.

Browser window would be even more unexplainable - but either way I'm afraid don't really understand how this is at all possible, [as the code processes the json](https://github.com/vesta-webtrees-2-custom-modules/vesta_shared_places/blob/1b5313be314a2983946a36e7ad78de4631e0dec2/resources/views/js/webtreesExt.phtml#L191) even in case of a non-success response code.

If this occurs again, please check the browser development tools for any console errors, and check where the json is displayed. 

Is this reproducible after deleting the newly created shared places, and re-creating them, or is there any other way to reproduce the error?

**Comment by Jefferson49** [comment]
> Is this reproducible after deleting the newly created shared places, and re-creating them, or is there any other way to reproduce the error?

I did not want to delete the data of the created shared places. Therefore, I tested with new shared places. 

**In a first test**, I got a weired result: I got logged out from webtrees and also got the "Unexpected token '<', " error at the same time.

The ["logged out" issue](https://www.webtrees.net/index.php/en/forum/help-for-release-2-1-x/37664-logged-out-in-problems?start=20#99665), is welknown, but the root cause is not understood. It seems that this was just random that it occured in this situation.

The unexpected token issue should be the one I mentioned above. However, it occured although I have already installed the bugfix.
![Unexpected token](https://github.com/vesta-webtrees-2-custom-modules/vesta_shared_places/assets/81484983/e2a68077-ad60-471e-895b-35d0e738aef3)

**In a second test**, everything just went fine without any problems.

It seems to be a very special behavior and occurs only from time to time. If the available data does not provide enough hints, it is not important to dive any deeper.
